plugin/plugin.go

```diff
@@ -317,6 +320,47 @@ func wrap_write_callback(ds *C.data_set_t, cvl *C.value_list_t, ud *C.user_data_
 	return 0
 }
 
+// First declare some types, interfaces, general functions
+
+// Shutters are objects that when called will shut down the plugin gracefully
+type Shutter interface {
+	Shutdown() error
+}
+
+// shutdownFuncs holds references to all shutdown callbacks
+var shutdownFuncs = make(map[string]Shutter)
+
+//export wrap_shutdown_callback
+func wrap_shutdown_callback() C.int {
+	if len(shutdownFuncs) <= 0 {
+		return -1
+	}
+	for n, s := range shutdownFuncs {
+		if err := s.Shutdown(); err != nil {
+			Errorf("The plugin named %s failed with error %v when called to shutdown", n, s)
+			return -1
+		}
+		break
```

Comment:
You should call all shutdown callbacks, not break after the first successful one.